apps/guide/src/content/02-creating-your-bot/03-command-handling.mdx

```diff
@@ -0,0 +1,220 @@
+
+<Alert title="Tip" type="success">
+For fully functional slash commands, there are three important pieces of code that need to be written. They are:
+
+1. The [individual command files](slash-commands), containing their definitions and functionality.
+2. The command handler, which dynamically reads the files and executes the commands.
+3. The [command deployment script](command-deployment), to register your slash commands with Discord so they appear in the interface.
+
+These steps can be done in any order, but **all are required** before the commands are fully functional.
+
+This page details how to complete **Step 2**. Make sure to also complete the other pages linked above!
+
+</Alert>
+
+## Loading command files
+
+Now that your command files have been created, your bot needs to load these files on startup.
+
+In your _`index.js`_ file, make these additions to the base template:
+
+<CH.Code>
+
+```js index.js focus=1:2,8
+import { readdirSync } from 'node:fs';
+import { join } from 'node:path';
+// focus[18:27]
+import { Client, Collection, Events, GatewayIntentBits } from 'discord.js';
+import config from './config.json';
+
+const client = new Client({ intents: [GatewayIntentBits.Guilds] });
+
+client.commands = new Collection();
+```
+
+</CH.Code>
+
+We recommend attaching a _`.commands`_ property to your client instance so that you can access your commands in other files. The rest of the examples in this guide will follow this convention. For TypeScript users, we recommend extending the base Client class to add this property, [casting](https://www.typescripttutorial.net/typescript-tutorial/type-casting/), or [augmenting the module type](https://www.typescriptlang.org/docs/handbook/modules.html#ambient-modules).
+
+<Alert title="Tip" type="success">
+	- The [_`fs`_](https://nodejs.org/api/fs.html) module is Node's native file system module. _`readdirSync`_ is used to
+	read the _`commands`_ directory and identify our command files. - The [_`path`_](https://nodejs.org/api/path.html)
+	module is Node's native path utility module. _`join`_ helps construct paths to access files and directories. One of
+	the advantages of _`path.join`_ is that it automatically detects the operating system and uses the appropriate
+	joiners. - The <DocsLink type="class" parent="Collection" /> class extends JavaScript's native
+	[_`Map`_](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) class, and includes
+	more extensive, useful functionality. _`Collection`_ is used to store and efficiently retrieve commands for execution.
+</Alert>
+
+Next, using the modules imported above, dynamically retrieve your command files with a few more additions to the _`index.js`_ file:
+
+<CH.Code>
+
+```js index.js mark=3:15 focus=3:15
+client.commands = new Collection();
+
+const commandsPath = join(__dirname, 'commands');
```

Comment:
`__dirname` is not defined in ESM.

Reply:
Alternatively, this can be used instead:
```suggestion
const commandsPath = new URL('./commands', import.meta.url);
```

Although this will require either adapting the rest of the code to use [`URL`](https://nodejs.org/dist/latest-v19.x/docs/api/url.html#new-urlinput-base)'s API, or use [`fileURLToPath`](https://nodejs.org/dist/latest-v19.x/docs/api/url.html#urlfileurltopathurl) without further modifications.